src/expr/synth_result.h

```diff
@@ -0,0 +1,82 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Encapsulation of the result of a synthesis query.
+ */
+
+#include "cvc5_public.h"
+
+#ifndef CVC5__EXPR__SYNTH_RESULT_H
+#define CVC5__EXPR__SYNTH_RESULT_H
+
+#include <iosfwd>
+#include <string>
+#include <vector>
+
+#include "expr/node.h"
+#include "util/result.h"
+
+namespace cvc5 {
+
+/**
+ * A result for a synthesis query. This can be used for synthesis, abduction,
+ * interpolation, and quantifier elimination.
+ */
+class SynthResult
+{
+ public:
+  enum Status
+  {
+    // the status has not been set
+    NONE,
+    // the synthesis query was successful, i.e. there is a solution
+    SUCCESS,
+    // the synthesis query resulted in failure, i.e. there is no solution
+    FAIL,
+    // the synthesis query is unknown, i.e. it is not know whether there is a
+    // solution.
+    UNKNOWN
+  };
+
+ public:
+  /** Default constructor */
+  SynthResult();
+  /** Constructor when the solution is successful */
+  SynthResult(const std::vector<Node>& sol);
+  /** Constructor when the solution is not successful */
+  SynthResult(
+      Status s,
+      Result::UnknownExplanation unknownExplanation = Result::UNKNOWN_REASON);
```

Comment:
Can we merge these two?

Reply:
Merge which?  

The Status and UnknownExplanation? No, since the unknown explanation should coincide with what is used in the (ordinary) Result class.

Reply:
No, the two constructors.

Reply:
You could, but there are combinations that don't make sense. If you have a solution, then your status (and unknown explanation) should always be determined, so you shouldnt have to provide it.